src/NServiceBus.RavenDB/Timeouts/TimeoutPersister.cs

```diff
@@ -65,45 +65,35 @@ public TimeoutPersister()
                 QueryHeaderInformation qhi;
                 using (var enumerator = session.Advanced.Stream(query, out qhi))
                 {
-                    if (qhi.TotalResults < 1000)
-                    {
-                        logger.DebugFormat("Retrieving {0} timeouts.", qhi.TotalResults);
-                    }
-                    else if (qhi.TotalResults < 10000)
-                    {
-                        logger.InfoFormat("Retrieving {0} timeouts. This could take some time.", qhi.TotalResults);
-                    }
-                    else
-                    {
-                        logger.WarnFormat("Retrieving {0} timeouts. This could take quite some time.", qhi.TotalResults);
-                    }
-
                     while (enumerator.MoveNext())
                     {
                         if (CancellationRequested())
                         {
                             return Enumerable.Empty<Tuple<string, DateTime>>();
                         }
 
-                        var dateTime = enumerator.Current.Document.Time;
-                        nextTimeToRunQuery = dateTime; // since results are sorted on time asc, this will get the max time < now
+                        results.Add(new Tuple<string, DateTime>(enumerator.Current.Document.Id, enumerator.Current.Document.Time));
+                    }
+                }
 
-                        if (dateTime > DateTime.UtcNow)
-                        {
-                            break; // break on first future timeout
-                        }
+                var nextTimeout = GetChunkQuery(session)
+                    .Where(t => t.Time > now)
+                    .Take(1)
+                    .FirstOrDefault();
 
-                        results.Add(new Tuple<string, DateTime>(enumerator.Current.Document.Id, dateTime));
-                    }
+                if (nextTimeout != null)
+                {
+                    nextTimeoutToExpire = nextTimeout.Time;
                 }
 
                 // Next execution is either now if we know we got stale results or at the start of the next chunk, otherwise we delay the next execution a bit
-                if (qhi.IsStale && results.Count == 0)
+                else if (qhi.IsStale && results.Count == 0)
```

Comment:
in the comment we say "if we know we got stale results **or** at the start of the next chunk" but the condition is **and**.